### PR TITLE
Add magnetic field forward modelling of rectangular prisms

### PR DESCRIPTION
**Pull request by santisoler**
Add forward modelling functions that calculate the magnetic fields of rectangular prisms. One function efficiently computes the three components of the magnetic field, while the other one calculates just one component. The functions make use of Choclo's forward modelling functions, while providing a nice interface, sanity checks and parallelization through Numba. Add examples and tests.

**Relevant issues/PRs:**

Related to #406

### CODE REVIEW

File: harmonica/_forward/prism_magnetic.py [hunk 1]
```diff
@@ -0,0 +1,427 @@
...
+    dtype=np.float64,
+    progressbar=False,
+    disable_checks=False,
+):
+    """
+    Magnetic field of right-rectangular prisms in Cartesian coordinates
+
+    Parameters
+    ----------
+    coordinates : list or 1d-array
```
Comment (santisoler) [inline harmonica/_forward/prism_magnetic.py:37]:
```suggestion
    coordinates : list of arrays
```

File: harmonica/_forward/prism_magnetic.py [hunk 2]
```diff
@@ -0,0 +1,427 @@
...
+        List or array containing ``easting``, ``northing`` and ``upward`` of
+        the computation points defined on a Cartesian coordinate system.
+        All coordinates should be in meters.
```
Comment (santisoler) [inline harmonica/_forward/prism_magnetic.py:40]:
```suggestion
    coordinates : list of arrays
        List of arrays containing the ``easting``, ``northing`` and ``upward`` coordinates of
        the computation points defined on a Cartesian coordinate system.
        All coordinates should be in meters.
```

File: harmonica/_forward/prism_magnetic.py [hunk 3]
```diff
@@ -0,0 +1,427 @@
...
+    prisms : list, 1d-array, or 2d-array
+        List or array containing the coordinates of the prism(s) in the
+        following order:
+        west, east, south, north, bottom, top in a Cartesian coordinate system.
+        All coordinates should be in meters. Coordinates for more than one
+        prism can be provided. In this case, *prisms* should be a list of lists
+        or 2d-array (with one prism per line).
```
Comment (santisoler) [inline harmonica/_forward/prism_magnetic.py:47]:
```suggestion
        or 2d-array (with one prism per row).
```

File: harmonica/tests/test_prism_magnetic.py [hunk 4]
```diff
@@ -0,0 +1,388 @@
...
+
+class TestAgainstChoclo:
+    """
+    Test forward modelling functions against dumb Choclo runs
+    """
+
+    @pytest.fixture()
+    def sample_prisms(self):
+        """
+        Return three sample prisms
```
Comment (santisoler) [inline harmonica/tests/test_prism_magnetic.py:281]:
```suggestion
        Return four sample prisms
```

**Review by LL-Geo — COMMENTED** [review]
Looks great! I'll fix a small typo, and ready to go 😄

**Comment by santisoler** [comment]
Awesome @LL-Geo! Thanks for the review and for the swift merge! 🚀

**Pull request merged**